compendium/ConfigurationAdmin/src/ConfigurationAdminImpl.cpp

```diff
@@ -734,35 +739,50 @@ ConfigurationAdminImpl::AddingService(
 	// update method. Return here without sending notification.
     logger->Log(cppmicroservices::logservice::SeverityLevel::LOG_DEBUG,
                   "New ManagedService with PID " + pid);
+
+    std::unordered_map<std::string, unsigned long> initialChangeCountByPid = {{pid, insertPair.first->second->GetChangeCount()}};
     return std::make_shared<
         TrackedServiceWrapper<cppmicroservices::service::cm::ManagedService>>(
-        pid, std::move(managedService));  
+            pid, std::move(initialChangeCountByPid), std::move(managedService));  
   }
   // Send a notification in case a valid configuration object 
   // was created before the service was active. The service's properties
   // need to be updated. 
   AnyMap properties{ AnyMap::UNORDERED_MAP_CASEINSENSITIVE_KEYS };
+  unsigned long initialChangeCount{0ul};
 
   if (it != std::end(configurations)) {
     try {
       properties = it->second->GetProperties();
+      initialChangeCount = it->second->GetChangeCount();
     } catch (const std::runtime_error&) {
       // Configuration is being removed
+      logger->Log(SeverityLevel::LOG_WARNING,
+                  "Attempted to update a configuration which has been removed.",
+                  std::current_exception());
     }
     // Only send notifications for configuration objects that have been
     // Updated. 
     if (it->second->HasBeenUpdatedAtLeastOnce()) {
       PerformAsync([this, pid, managedService, properties] {
         notifyServiceUpdated(pid, *managedService, properties, *logger);
```

Comment:
Is this bypassing the new mechanism in the managedServiceWrapper by calling notifyServiceUpdated directly the change count in the managedServiceWrapper doesn't get checked or incremented. I think there might be other cases in the code where notifyServiceUpdated and notifyServiceRemoved are called directly.

Reply:
managedServiceWrapper doesn't exist at this point since `AddingService` hasn't returned yet. So I don't think it's bypassing the new mechanism. When the managedServiceWrapper is returned, it will contain the change count at the point in which notifyServiceUpdated was called.